src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1436,40 +1529,33 @@ export const collections = { blog };
 
 This function accepts the following properties:
 
-#### `type`
+#### `loader`
 
 <p>
 
-**Type:** `'content' | 'data'`<br />
-**Default:** `'content'`<br />
-<Since v="2.5.0" />
+**Type:** <code>() => Promise&lt;Array&lt;\{ id: string, [key: string]: any }&gt; | Record&lt;string, Record&lt;string, any&gt;&gt;&gt; | <a href="/en/reference/loader-reference/#loader-api">Loader</a></code>
+<Since v="5.0.0" />
 </p>
 
-`type` is a string that defines the type of entries stored within a collection:
-
-- `'content'` - for content-authoring formats like Markdown (`.md`), MDX (`.mdx`), or Markdoc (`.mdoc`)
-- `'data'` - for data-only formats like JSON (`.json`) or YAML (`.yaml`)
-
-:::tip
-This means collections **cannot** store a mix of content and data formats. You must split these entries into separate collections by type.
-:::
+[See the `Content Collection` guide](/en/guides/content-collections/#defining-the-collection-loader) for example usage.
```

Comment:
Looks good to me! I'm just wondering if, to be consistent, we shouldn't make a short description before referring to the guide. Other entries usually have at least one sentence. Something like:

```suggestion
A `loader` is either an object or a function that allows you to load data from any source, local or remote, into content collections.

[See the `Content Collection` guide](/en/guides/content-collections/#defining-the-collection-loader) for example usage.
```